core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentGenerator.java

```diff
@@ -324,7 +325,9 @@ private void writeStartRaw(String fieldName) throws IOException {
     public void writeEndRaw() {
         assert base != null : "JsonGenerator should be of instance GeneratorBase but was: " + generator.getClass();
         if (base != null) {
-            base.getOutputContext().writeValue();
+            JsonStreamContext context = base.getOutputContext();
+            assert (context instanceof JsonWriteContext) : "Expected an instance of JsonWriteContext but was: " + context.getClass();
+            ((JsonWriteContext) context).writeValue();
```

Comment:
what i meant was, before we had `base.getOutputContext().writeValue()`, which will call the `writeValue()` method on whatever sub-class the `base.getOutputContext()` resolves to, so do we really need to do the cast?  Wouldn't `base.getOutputContext().writeValue()` call the `writeValue()` of `JsonWriteContext` if `base.getOutputContext()` returns an instance of `JsonWriteContext`?